src/group_theory/group_action/defs.lean

```diff
@@ -249,6 +267,23 @@ a multiplicative action of `N` on `α`. -/
 an additive action of `N` on `α`. -/
 add_decl_doc add_action.comp_hom
 
+variable {α}
+
+/-- The tautological action by `function.End α` on `α`. -/
+instance function_End : mul_action (function.End α) α :=
+{ smul := ($),
+  one_smul := λ _, rfl,
+  mul_smul := λ _ _ _, rfl }
```

Comment:
Do we have this instance for `perm α` on `α`?

Reply:
I don't think so. I defined this instance only to get `mul_action.of_End_hom`, and it doesn't seem like there is a `mul_action.of_perm_hom` either.